connectd/tor_autoservice.c

```diff
@@ -133,6 +152,54 @@ static struct wireaddr *make_onion(const tal_t *ctx,
 		      "Tor didn't give us a ServiceID");
 }
 
+static struct wireaddr *make_fixed_onion(const tal_t *ctx,
+				   struct rbuf *rbuf,
+				   const struct wireaddr *local, const char *blob, u16 port)
+{
+	char *line;
+	struct wireaddr *onion;
+	char *blob64;
+
+
+	blob64 = b64_encode(tmpctx,(char *) blob, 64);
+
+	tor_send_cmd(rbuf,
+			  tal_fmt(tmpctx, "ADD_ONION ED25519-V3:%s Port=%d,%s Flags=DiscardPK",
+					blob64, port, fmt_wireaddr(tmpctx, local)));
+
+	while ((line = tor_response_line_wfail(rbuf)) != NULL) {
+		const char *name;
+		if (line && strstarts(line, "Onion address collision"))
```

Comment:
Don't need to check if `line` is NULL, since loop would have exited.

Reply:
;-)